### PR TITLE
Readthedocs

### PR DESCRIPTION
**Pull request by jspaaks**
This PR adds automatic generation of api docs on readthedocs (https://matchms.readthedocs.io/en/latest/) using sphinxcontrib.apidoc https://pypi.org/project/sphinxcontrib-apidoc/

### CODE REVIEW

File: readthedocs/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+sphinx==3.*
+sphinxcontrib-apidoc
+numpy
```
Comment (fdiblen) [inline readthedocs/requirements.txt:3]: Do we really need numpy and matplotlib? :roll_eyes:

File: readthedocs/Makefile [hunk 2]
```diff
@@ -1,27 +1,20 @@
 # Minimal makefile for Sphinx documentation
```
Comment (fdiblen) [inline readthedocs/Makefile:1]: This is a better Makefile:
https://github.com/readthedocs/readthedocs.org/blob/master/docs/Makefile

File: readthedocs/make.bat [hunk 3]
```diff
@@ -1,36 +1,35 @@
...
-
-%SPHINXBUILD% -M %1 %SOURCEDIR% %BUILDDIR% %SPHINXOPTS%
-goto end
-
-:help
-%SPHINXBUILD% -M help %SOURCEDIR% %BUILDDIR% %SPHINXOPTS%
-
-:end
-popd
+@ECHO OFF
```
Comment (fdiblen) [inline readthedocs/make.bat:1]: this is for the bat file:
https://github.com/readthedocs/readthedocs-build/blob/master/integration_tests/minimal_project/make.bat

File: readthedocs/requirements.txt (hunk 1 above)
Comment (jspaaks) [reply readthedocs/requirements.txt:3]: Yes, we do (i do not understand why). Something to look into later I guess

File: readthedocs/Makefile (hunk 2 above)
Comment (jspaaks) [reply readthedocs/Makefile:1]: better how?

File: readthedocs/source/index.rst [hunk 4]
```diff
@@ -0,0 +1,16 @@
+Welcome to Getting Started with Sphinx's documentation!
```
Comment (fdiblen) [inline readthedocs/source/index.rst]: Sphinx's documentation --> matchms's documentation

File: readthedocs/source/index.rst [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+=======================================================
+
+.. automodule:: matchms
+   :members:
+
+.. toctree::
+   :maxdepth: 2
```
Comment (fdiblen) [inline readthedocs/source/index.rst]: shall we change maxdepth to 3?

File: readthedocs/make.bat (hunk 3 above)
Comment (jspaaks) [reply readthedocs/make.bat:1]: I used some tools to automatically generate the skeleton. I find the ``conf.py``, ``Makefile``, and ``make.bat`` very difficult to understand to the point where I don't dare to touch them.

**Review by fdiblen — CHANGES_REQUESTED** [review]
I suggested few minor changes

File: readthedocs/source/index.rst (hunk 4 above)
Comment (jspaaks) [reply readthedocs/source/index.rst]: ~~myea later~~ done

File: readthedocs/source/index.rst (hunk 5 above)
Comment (jspaaks) [reply readthedocs/source/index.rst]: Sure

**Comment by jspaaks** [comment]
Hey cool sonarcloud

**Comment by jspaaks** [comment]
Not perfect by any means, but I've had it with readthedocs, sphinx and sphinxcontrib-apidoc

File: readthedocs/Makefile (hunk 2 above)
Comment (fdiblen) [reply readthedocs/Makefile:1]: extra build options such as generating pdf and some other formats. Not a big deal for now though.

**Review by fdiblen — APPROVED** [review]

**Pull request merged**